subsys/bluetooth/host_extensions/conn_evt_cb.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/hci.h>
+#include <bluetooth/evt_cb.h>
+#include <helpers/nrfx_gppi.h>
+#include <bluetooth/hci_vs_sdc.h>
+
+#include <nrfx_egu.h>
+
+#if defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU0)
+#define EGU_INST_IDX 0
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU1)
+#define EGU_INST_IDX 1
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU2)
+#define EGU_INST_IDX 2
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU3)
+#define EGU_INST_IDX 3
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU4)
+#define EGU_INST_IDX 4
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU5)
+#define EGU_INST_IDX 5
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU020)
+#define EGU_INST_IDX 020
+#else
+#error "Unknown EGU instance"
+#endif
+
+static nrfx_egu_t egu_inst = NRFX_EGU_INSTANCE(EGU_INST_IDX);
+
+struct conn_evt_cb_cfg {
+	struct bt_conn_evt_cb cb;
+	struct bt_conn *conn;
+	void *user_data;
+	uint32_t prepare_time_us;
+
+	uint8_t ppi_channel;
+	uint16_t conn_interval_us;
```

Comment:
Can we just read it from `bt_conn_get_info()` instead of storing it?

Reply:
We can. I have no strong preference of how to obtain it. Caching it seemed like the simpler option